internal/devconfig/config.go

```diff
@@ -346,9 +347,18 @@ func (c *Config) NixPkgsCommitHash() string {
 func (c *Config) Env() map[string]string {
 	env := map[string]string{}
 	for _, i := range c.included {
-		maps.Copy(env, i.Env())
+		// Here, in new pluginPath we replace the keyword "$PATH"
+		// with what we have for PATH so far. If so far we have /plugin1/bin:$PATH
+		// and new plugin adds /plugin2/bin/:$PATH the result becomes
+		// /plugin2/bin/:/plugin1/bin/:$PATH
+		pluginEnv := mergePATHsFromTwoEnvs(env, i.Env())
+		maps.Copy(env, pluginEnv)
 	}
-	maps.Copy(env, c.Root.Env)
+	// Here we merge the processed PATH from all plugins to PATH in config env.
+	// So if config env has /config/env/:$PATH merging with the plugin PATH
+	// it becomes /config/env/:/plugin2/bin/:/plugin1/bin/:$PATH
+	rootEnv := mergePATHsFromTwoEnvs(env, c.Root.Env)
+	maps.Copy(env, rootEnv)
```

Comment:
Can you use `OSExpandEnvMap` instead? 

Basically take the environment provided by the plugins and use it to expand the environment of the config.

Reply:
@mikeland73 after your suggestion, I tried it and it it doesn't work. The issue with `OSExpandEnvMap` is it merges the two PATHs correctly but it applies the (non-nix) system PATH to the `:$PATH` at the end. So with `OSExpandEnvMap` PATH becomes: `/my/config:/my/plugin:/opt/homebrew/sbin:/usr/local/bin:...`
with `mergePATHsFromTwoEnvs` PATH becomes: `/my/config:/my/plugin:/Users/mohsenansari/code/go.jetpack.io/devbox/examples/stacks/rails/.devbox/nix/profile/default/bin:...:/opt/homebrew/sbin:/usr/local/bin:...`

So OSExpandEnvMap can't work here because I don't want to expand $PATH env yet, I just want to merge them correctly without losing `:$PATH` suffix or `$PATH:` prefix to it.

Reply:
Would something like 

```golang
func OSExpandIfPossible(env, existingEnv map[string]string) map[string]string {
	mapping := func(value string) string {
		// If the value is not set in existingEnv, return the value wrapped in ${...}
		if existingEnv == nil || existingEnv[value] == "" {
			return fmt.Sprintf("${%s}", value)
		}
		return existingEnv[value]
	}

	res := map[string]string{}
	for k, v := range env {
		res[k] = os.Expand(v, mapping)
	}
	return res
}
```

work?

Reply:
And some unit tests if you decide to go this route:

```golang
package conf

import "testing"

func TestOSExpandIfPossible(t *testing.T) {
	tests := []struct {
		name        string
		env         map[string]string
		existingEnv map[string]string
		want        map[string]string
	}{
		{
			name: "basic expansion",
			env: map[string]string{
				"FOO": "$BAR",
				"BAZ": "${QUX}",
			},
			existingEnv: map[string]string{
				"BAR": "bar_value",
				"QUX": "qux_value",
			},
			want: map[string]string{
				"FOO": "bar_value",
				"BAZ": "qux_value",
			},
		},
		{
			name: "missing values remain as template",
			env: map[string]string{
				"FOO": "$BAR",
				"BAZ": "${QUX}",
			},
			existingEnv: map[string]string{
				"BAR": "bar_value",
				// QUX is missing
			},
			want: map[string]string{
				"FOO": "bar_value",
				"BAZ": "${QUX}",
			},
		},
		{
			name: "nil existing env",
			env: map[string]string{
				"FOO": "$BAR",
				"BAZ": "${QUX}",
			},
			existingEnv: nil,
			want: map[string]string{
				"FOO": "${BAR}",
				"BAZ": "${QUX}",
			},
		},
		{
			name: "empty existing env",
			env: map[string]string{
				"FOO": "$BAR",
			},
			existingEnv: map[string]string{},
			want: map[string]string{
				"FOO": "${BAR}",
			},
		},
		{
			name: "mixed literal and variable",
			env: map[string]string{
				"FOO": "prefix_${BAR}_suffix",
			},
			existingEnv: map[string]string{
				"BAR": "bar_value",
			},
			want: map[string]string{
				"FOO": "prefix_bar_value_suffix",
			},
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got := OSExpandIfPossible(tt.env, tt.existingEnv)
			if len(got) != len(tt.want) {
				t.Errorf("OSExpandIfPossible() got %v entries, want %v entries", len(got), len(tt.want))
			}
			for k, v := range tt.want {
				if got[k] != v {
					t.Errorf("OSExpandIfPossible() for key %q = %q, want %q", k, got[k], v)
				}
			}
		})
	}
}
```

Reply:
`OSExpandIfPossible` works. I included this suggestion along with the unit test and added a case for path to the unit test as well.